spec/requests/vms_spec.rb

```diff
@@ -1621,6 +1621,50 @@ def query_match_regexp(*tables)
     end
   end
 
+  context "Vm set_description action" do
+    it "to an invalid vm" do
+      api_basic_authorize action_identifier(:vms, :rename)
+
+      post(invalid_vm_url, :params => gen_request(:set_description, "new_description" => "test"))
+
+      expect(response).to have_http_status(:not_found)
+    end
+
+    it "to an invalid vm without appropriate role" do
+      api_basic_authorize
+
+      post(invalid_vm_url, :params => gen_request(:set_description, "new_description" => "test"))
+
+      # this is not found because you can't find the vm to set the description
+      # rather than you can find the vm but you can't do the operation to i
+      expect(response).to have_http_status(:not_found)
```

Comment:
How is this action different from e.g. VM Retire which returns `:forbidden` for this exact test: https://github.com/ManageIQ/manageiq-api/blob/1372260a328c9b16339bfd0e019c0b8a5fc0fe59/spec/requests/vms_spec.rb#L1339-L1345

I don't have an issue with returning 404 instead of 403 I just think we need to be consistent